### ISSUE TITLE
2 failures and 1 error while performing the tests

### ISSUE THREAD

**Issue by ashkurti** [MDAnalysis/mdanalysis#551]
Hi MDAnalysis developers!!

Any advice on the following error and failures will be much appreciated. I am aiming to get everything working before contributing to the MDAnalysis code with some analysis enhancements.
```
======================================================================
ERROR: MDAnalysisTests.coordinates.test_trz.TestTRZWriter2.test_writer_trz_from_other
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/ardita/MDAPCAvenv/lib/python2.7/site-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "MDAnalysisTests/coordinates/test_trz.py", line 189, in test_writer_trz_from_other
    u2 = mda.Universe(two_water_gro, outfile)
  File "/users/ardita/MDAPCAvenv/lib/python2.7/site-packages/MDAnalysis-0.13.0_dev0-py2.7-linux-x86_64.egg/MDAnalysis/core/AtomGroup.py", line 3791, in __init__
    "Error: {2}".format(self.filename, parser, err))
IOError: Failed to load from the topology file MDAnalysisTests/data/two_water_gro.gro with parser <class 'MDAnalysis.topology.GROParser.GROParser'>.
Error: [Errno 5] Cannot open file or stream in mode='r'.: "'MDAnalysisTests/data/two_water_gro.gro'"
-------------------- >> begin captured stderr << ---------------------
/users/ardita/MDAPCAvenv/lib/python2.7/site-packages/MDAnalysis-0.13.0_dev0-py2.7-linux-x86_64.egg/MDAnalysis/coordinates/TRZ.py:539: UserWarning: Timestep didn't have the following attributes: 'pressure, pressure_tensor, total_energy, potential_energy, kinetic_energy, temperature, step', these will be set to 0 in the output trajectory
  "".format(", ".join(faked_attrs)))
/users/ardita/MDAPCAvenv/lib/python2.7/site-packages/MDAnalysis-0.13.0_dev0-py2.7-linux-x86_64.egg/MDAnalysis/coordinates/TRZ.py:547: UserWarning: Timestep didn't have velocity information, this will be set to zero in output trajectory. 
  warnings.warn("Timestep didn't have velocity information, "

--------------------- >> end captured stderr << ----------------------
-------------------- >> begin captured logging << --------------------
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading MDAnalysisTests/data/two_water_gro.gro...
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: test_write_trajectory (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "MDAnalysisTests/coordinates/test_netcdf.py", line 139, in test_write_trajectory
    'ncdf coord output not float32')
AssertionError: ncdf coord output not float32
-------------------- >> begin captured logging << --------------------
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading MDAnalysisTests/data/Amber/bala.ncdf...
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /tmp/tmp2QChD0/ncdf-writer-1.ncdf...
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: test_write_trajectory (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriterTZ2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "MDAnalysisTests/coordinates/test_netcdf.py", line 139, in test_write_trajectory
    'ncdf coord output not float32')
AssertionError: ncdf coord output not float32
-------------------- >> begin captured logging << --------------------
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading MDAnalysisTests/data/Amber/tz2.truncoct.nc...
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /tmp/tmpdfdaTZ/ncdf-writer-1.ncdf...
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 3275 tests in 983.183s

FAILED (KNOWNFAIL=3, errors=1, failures=2)
```

**Comment by orbeckst** [comment]
"AssertionError: ncdf coord output not float32" looks like something that was actually fixed recently #518 so it makes me think that what you're testing might be an older installation; at least that happens to me occasionally. Can you completely wipe all MDAnalysis installations and reinstall freshly?

**Comment by richardjgowers** [comment]
Hello!

The first error looks like it's not opening the file, which might be related to install too.

Either way, these aren't really that important unless you're using either TRZ or NCDF formats.

**Comment by ashkurti** [comment]
@orbeckst @richardjgowers thanks for your answers. I have reinstalled and doing the tests again now. It takes approx 900+seconds to execute all the tests.

The procedure that I have followed for this is:
```
git clone https://github.com/MDAnalysis/mdanalysis.git
cd mdanalysis 
cd package
python setup.py install
cd ../testsuite
python setup.py install
```
and then:
```
python -c 'from MDAnalysis.tests import test; test(argv=["--exe", "-v"])'
```
or
```
nosetests --verbosity=3
```

Is this the correct process? I have installed all of HDF5, netCDF4 and netcdf4-python therefore would expect not to have any errors for NCDF formats. Am I wrong ...

Will let you know when the new round of testing is over :)

**Comment by ashkurti** [comment]
Ok, tests are over now. At least the error disappeared but still get the following:
```
======================================================================
FAIL: test_write_trajectory (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/coordinates/test_netcdf.py", line 139, in test_write_trajectory
    'ncdf coord output not float32')
AssertionError: ncdf coord output not float32
-------------------- >> begin captured logging << --------------------
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/Amber/bala.ncdf...
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /tmp/tmpA1K0OP/ncdf-writer-1.ncdf...
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: test_write_trajectory (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriterTZ2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/coordinates/test_netcdf.py", line 139, in test_write_trajectory
    'ncdf coord output not float32')
AssertionError: ncdf coord output not float32
-------------------- >> begin captured logging << --------------------
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/Amber/tz2.truncoct.nc...
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /tmp/tmpA2I5XQ/ncdf-writer-1.ncdf...
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 3275 tests in 985.603s

FAILED (SKIP=3, failures=2)
```

**Comment by richardjgowers** [comment]
`python -c 'from MDAnalysis.tests import test; test(argv=["--exe", "-v", "--parallel-processes=4", "--process-timeout=120"])'`

This should let you run the tests in parallel, which helps make it less painful

There's also ways to run specific tests:
https://github.com/MDAnalysis/mdanalysis/wiki/UnitTests#developers

**Comment by tylerjereddy** [comment]
@ashkurti I wrote that test fairly recently, so it does suggest that the latest version of the development branch isn't being used to run the tests. Could just compare the commit hash of the latest branch on the command line with github or the last git log entry, etc. Maybe make sure you're on the development branch instead of master--although that's fairly obvious I suppose (sometimes defaults to master on checkout?--but I think @orbeckst changed the github default to develop).

**Comment by ashkurti** [comment]
Yes, the default branch is develop. 

Would the following help in terms of commit:
```
[MDAPCAvenv] ardita@tirith 108% git log
commit 0c3aa90cf0a748642a9aab5e50562ca1b9fedc18
Merge: 82ce412 4fa0e65
Author: kain88-de <kain88-de@users.noreply.github.com>
Date:   Mon Nov 23 20:35:18 2015 +0100

    Merge pull request #549 from MDAnalysis/issue-548

    Issue 548

commit 4fa0e6569ee5ceb7f1d76f4cb8bf53e75dfe14a1
Author: Richard Gowers <richardjgowers@gmail.com>
Date:   Mon Nov 23 10:31:37 2015 -0700

    Fixes for Issue #548

    GROParser now doesn't read infinitely, but only the number of atoms
    stated in the top of the file.

commit 09c03f46fc64156c4c62b3795fb9dcd3137b1171
Author: Richard Gowers <richardjgowers@gmail.com>
Date:   Mon Nov 23 10:23:00 2015 -0700

    Added tests for Issue #548
:
commit 0c3aa90cf0a748642a9aab5e50562ca1b9fedc18
Merge: 82ce412 4fa0e65
Author: kain88-de <kain88-de@users.noreply.github.com>
Date:   Mon Nov 23 20:35:18 2015 +0100

    Merge pull request #549 from MDAnalysis/issue-548

    Issue 548

commit 4fa0e6569ee5ceb7f1d76f4cb8bf53e75dfe14a1
Author: Richard Gowers <richardjgowers@gmail.com>
Date:   Mon Nov 23 10:31:37 2015 -0700

    Fixes for Issue #548

    GROParser now doesn't read infinitely, but only the number of atoms
    stated in the top of the file.

commit 09c03f46fc64156c4c62b3795fb9dcd3137b1171
Author: Richard Gowers <richardjgowers@gmail.com>
Date:   Mon Nov 23 10:23:00 2015 -0700

    Added tests for Issue #548

commit 82ce412a0d44c1a10ade4f698b80acb73fd31fc6
Merge: 06ec4b6 af37201
Author: kain88-de <kain88-de@users.noreply.github.com>
Date:   Thu Nov 19 19:55:10 2015 +0100

    Merge pull request #547 from MDAnalysis/feature-betterdata

    Feature betterdata

commit af372016f08134ea66d42cdfa3f931174bac5e74
Author: Richard Gowers <richardjgowers@gmail.com>
Date:   Wed Nov 18 21:35:47 2015 -0800

Broken pipe
[MDAPCAvenv] ardita@tirith 109% git log
commit 0c3aa90cf0a748642a9aab5e50562ca1b9fedc18
Merge: 82ce412 4fa0e65
Author: kain88-de <kain88-de@users.noreply.github.com>
Date:   Mon Nov 23 20:35:18 2015 +0100

    Merge pull request #549 from MDAnalysis/issue-548

    Issue 548

commit 4fa0e6569ee5ceb7f1d76f4cb8bf53e75dfe14a1
Author: Richard Gowers <richardjgowers@gmail.com>
Date:   Mon Nov 23 10:31:37 2015 -0700

    Fixes for Issue #548

    GROParser now doesn't read infinitely, but only the number of atoms
    stated in the top of the file.

commit 09c03f46fc64156c4c62b3795fb9dcd3137b1171
Author: Richard Gowers <richardjgowers@gmail.com>
Date:   Mon Nov 23 10:23:00 2015 -0700

    Added tests for Issue #548

commit 82ce412a0d44c1a10ade4f698b80acb73fd31fc6
Merge: 06ec4b6 af37201
Author: kain88-de <kain88-de@users.noreply.github.com>
Date:   Thu Nov 19 19:55:10 2015 +0100

    Merge pull request #547 from MDAnalysis/feature-betterdata

    Feature betterdata

commit af372016f08134ea66d42cdfa3f931174bac5e74
Author: Richard Gowers <richardjgowers@gmail.com>
Date:   Wed Nov 18 21:35:47 2015 -0800

:
```

**Comment by hainm** [comment]
just have comment here: it's very weird that MDA developers prefer 'develop' branch over 'master' branch. For most github repos I visited, 'master' is actually 'develop'.

**Comment by richardjgowers** [comment]
@ashkurti looks like you're installing everything properly.  It might be a real bug.. what's your scipy version? (eg below)

``` python
In [1]: import scipy

In [2]: scipy.__version__
Out[2]: '0.16.0'

```

**Comment by ashkurti** [comment]
@richardjgowers 
```
[MDAPCAvenv] ardita@tirith 112% python -c "import scipy; print scipy.__version__"
0.16.1
```

**Comment by orbeckst** [comment]
I think you have the right branch and an up-to-date environment. I have scipy 0.16.1, too.

I just did the following under Ubuntu 14.04:

```
rm -r ~/.local/lib/python2.7/site-packages/MDAnalysis*
git pull
git checkout develop
./maintainer/install_all.sh -u
cd testsuite/MDAnalysisTests
./mda_nosetests -v --processes=24 --process-timeout=120 --with-memleak
```

... and  I got on 12 cores + 12 hyperthreading
```
Ran 3360 tests in 68.769s
OK (KNOWNFAIL=3, SKIP=2)
```

Everything is wonderful.
* The SKIP=2 turn into PASS when I add `hole` and `clustalw2` to my environment.
* all tests pass, including the analysis tests #552 

However, I then just ran everything *again* and within 77s I got
```
FAILED (KNOWNFAIL=3, SKIP=2, error=1)
```

The `error=1` is curious because it is spurious (when I did exactly the same thing 2 minutes earlier,  there were no errors. The one error was

```
======================================================================
ERROR: MDAnalysisTests.coordinates.test_xyz.TestXYZReader.test_get_writer_container
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/usr/local/lib/python2.7/dist-packages/nose/case.py", line 197, in runTest
    self.test(*self.arg)
  File "/nfs/homes2/oliver/.local/lib/python2.7/site-packages/MDAnalysisTests-0.13.0.dev0-py2.7.egg/MDAnalysisTests/coordinates/base.py", line 198, in test_get_writer_container
    with tempdir.in_tempdir():
  File "/usr/lib/python2.7/contextlib.py", line 17, in __enter__
    return self.gen.next()
  File "build/bdist.linux-x86_64/egg/tempdir.py", line 44, in in_tempdir
    old_path = os.getcwd()
OSError: [Errno 2] No such file or directory
```

This might be related to the mysterious failures that [I just complained about](https://groups.google.com/d/msg/mdnalysis-devel/dP4N0F8nhlo/gUrfxSoRCgAJ) on the developer mailinglist.

So there's something fishy with at least some of the tests. 

1. Did you rerun your tests and do you get the same failures?
2. Did you completely wipe your old installation (or use a clean virtualenv)?

Finally, I very much recommend using multiple processes for the testing. Also, for this issue you should be able to just run

```
./mda_nosetests -v --processes=4 --process-timeout=120 --with-memleak ./coordinates/test_netcdf.py
```

which takes *much* less time than running everything and only contains your failing tests. Ideally you want to see
```
Ran 42 tests in 2.707s

OK
```

**Comment by hainm** [comment]
it's better to run the test in single core to debug.

 How's about
```bash
cd testsuite/MDAnalysisTests/coordinates
python test_netcdf.py
```
(or update nosetests command)

**Comment by orbeckst** [comment]
Apologies to @ashkurti for hijacking the issue: I opened #556 and we continue discussing the spurious test failures there and focus *this* issue on the problem that you are having. If it turns out that it is related then we might be able to kill two birds with one stone but right now I am assuming that #551 and #556 are different.

**Comment by ashkurti** [comment]
@orbeckst and @hainm thanks for your tips.

I ran again now: 
```
./mda_nosetests -v --with-memleak ./coordinates/test_netcdf.py
```
And I get two failures. The output produced is this:
```
[MDAPCAvenv] ardita@moriarty 390% ./mda_nosetests -v --with-memleak ./coordinates/test_netcdf.py
test_amber_proteinselection (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
test_full_slice (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
test_initial_frame_is_0 (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
test_load_prm (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
test_metadata (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
test_n_atoms (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
test_n_frames (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
test_periodic (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
test_rewind (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
test_slice_iteration (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
Test that coordinate arrays are filled as soon as the trajectory ... ok
test_sum_centres_of_geometry (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReader) ... ok
Check forces on first frame ... ok
Check forces on second frame ... ok
Check positions on first frame ... ok
Check positions on second frame ... ok
Check time on first frame ... ok
Check time on second frame ... ok
test_amber_proteinselection (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
test_full_slice (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
test_initial_frame_is_0 (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
test_load_prm (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
test_metadata (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
test_n_atoms (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
test_n_frames (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
test_periodic (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
test_rewind (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
test_slice_iteration (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
Test that coordinate arrays are filled as soon as the trajectory ... ok
test_sum_centres_of_geometry (MDAnalysisTests.coordinates.test_netcdf.TestNCDFReaderTZ2) ... ok
test_OtherWriter (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriter) ... ok
test_TRR2NCDF (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriter) ... ok
test to write NCDF from AtomGroup (Issue 116) ... ok
test_write_trajectory (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriter) ... FAIL
test_OtherWriter (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriterTZ2) ... ok
test_TRR2NCDF (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriterTZ2) ... ok
test to write NCDF from AtomGroup (Issue 116) ... ok
test_write_trajectory (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriterTZ2) ... FAIL
test_pos (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriterVelsForces) ... ok
test_pos_force (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriterVelsForces) ... ok
test_pos_vel (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriterVelsForces) ... ok
test_pos_vel_force (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriterVelsForces) ... ok

======================================================================
FAIL: test_write_trajectory (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriter)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/coordinates/test_netcdf.py", line 139, in test_write_trajectory
    'ncdf coord output not float32')
AssertionError: ncdf coord output not float32
-------------------- >> begin captured logging << --------------------
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/Amber/bala.ncdf...
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /tmp/tmp_zaQSW/ncdf-writer-1.ncdf...
--------------------- >> end captured logging << ---------------------

======================================================================
FAIL: test_write_trajectory (MDAnalysisTests.coordinates.test_netcdf.TestNCDFWriterTZ2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/coordinates/test_netcdf.py", line 139, in test_write_trajectory
    'ncdf coord output not float32')
AssertionError: ncdf coord output not float32
-------------------- >> begin captured logging << --------------------
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/Amber/tz2.truncoct.nc...
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /tmp/tmpkfCo6j/ncdf-writer-1.ncdf...
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 42 tests in 15.233s

FAILED (failures=2)
```

**Comment by hainm** [comment]
I've just run tests with 

`nosetests -vs .`

and got `OSError: [Errno 2] No such file or directory` error too. It's likely related to `tempdir`

**Comment by orbeckst** [comment]
Thanks @hainm although I still think that this is related to #556.

@ashkurti can you try the following and report the results?
```python
import numpy as np
import MDAnalysis as mda
import netCDF4

from MDAnalysis.tests.datafiles import PRMncdf, NCDF

dataset = netCDF4.Dataset(NCDF, "r", format="NETCDF3")
coor = dataset.variables['coordinates']
print(coor.dtype)
print(coor.dtype == np.float32)
```

And then what the test tries to do: first write a new ncdf trajectory and then read it again:
```python
u = mda.Universe(PRMncdf, NCDF)
with mda.Writer("new.ncdf", u.atoms.n_atoms) as W:
   for ts in u.trajectory:
        W.write(ts)

newdata = netCDF4.Dataset("new.ncdf", "r", format="NETCDF3")
newcoor = newdata.variables['coordinates']
print(newcoor.dtype)
print(newcoor.dtype == np.float32)
```

**Comment by kain88-de** [comment]
Tempdir hasn't been updated in 2 years why should it suddenly fail. There must be another issue.

**Comment by hainm** [comment]
> Tempdir hasn't been updated in 2 years why should it suddenly fail. There must be another issue.

I don't see two things related.

**Comment by kain88-de** [comment]
It's just done. Nothing much you can do on a 30-50 line module.

**Comment by orbeckst** [comment]
@hainm and @kain88-de : could we please move the (non)discussion about `tempdir` to #556?

Thanks!

**Comment by kain88-de** [comment]
@ashkurti  Are you sure about the version that you have installed and which is being tested?
run `python -c `import MDAnalysis; print(MDAnalysis.__file__, MDAnalysis.__version__)'

**Comment by ashkurti** [comment]
@orbeckst This is what I get with the code lines you suggested (so no problems appear with that):
```
>>> import numpy as np
>>> import MDAnalysis as mda
>>> import netCDF4
>>> from MDAnalysis.tests.datafiles import PRMncdf, NCDF
>>> dataset = netCDF4.Dataset(NCDF, "r", format="NETCDF3")
>>> coor = dataset.variables['coordinates']
>>> print(coor.dtype)
float32
>>> print(coor.dtype == np.float32)
True
>>> u = mda.Universe(PRMncdf, NCDF)
>>> with mda.Writer("new.ncdf", u.atoms.n_atoms) as W:
...     for ts in u.trajectory:
...             W.write(ts)
... 
>>> newdata = netCDF4.Dataset("new.ncdf", "r", format="NETCDF3")
>>> newcoor = newdata.variables['coordinates']
>>> print(newcoor.dtype)
float32
>>> print(newcoor.dtype == np.float32)
True
```

**Comment by ashkurti** [comment]
@kain88-de The versions I am using are the following:
```
python -c 'import MDAnalysis; print(MDAnalysis.__file__, MDAnalysis.__version__)'
('/users/ardita/MDAPCAvenv/lib/python2.7/site-packages/MDAnalysis-0.13.0_dev0-py2.7-linux-x86_64.egg/MDAnalysis/__init__.pyc', '0.13.0-dev0')

```

**Comment by ashkurti** [comment]
I noticed that there is at least one new commit due to the last few hours here. And since there was a hint in #556 that there might be some things solved, I pulled everything again, and rerun the tests with:
./mda_nosetests -v --processes=8 --with-memleak

I did this twice, noticed a different number of errors each time and then I repeated it a third time. The first time, I only got 300 errors. The second time got 345 errors. The third time got 417 errors. I thought the number of failures/errors was supposedly deterministic. 
You can find part of the log of my second try at: https://gist.github.com/ashkurti/b2347d8d3f71542ec95f
and the full log of my third try at: https://gist.github.com/ashkurti/f87c56693e41a2f80948

**Comment by ashkurti** [comment]
However, I wanted to integrate some code enhancements. Should I then just ignore whether I have any errors for other tests as long as I do not have errors for my own tests and then submit a pull request? Thanks.

**Comment by ashkurti** [comment]
Now, I used the nosetests command for the tests and only got three errors: 
```
nosetests --verbosity=3
```
```
======================================================================
ERROR: test_hausdorff_bound (MDAnalysisTests.analysis.test_psa.TestPSAnalysis)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/analysis/test_psa.py", line 57, in tearDown
    shutil.rmtree('psadata') # remove psa data directory
  File "/usr/remote/python/2.7.8/lib/python2.7/shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "/usr/remote/python/2.7.8/lib/python2.7/shutil.py", line 256, in rmtree
    onerror(os.rmdir, path, sys.exc_info())
  File "/usr/remote/python/2.7.8/lib/python2.7/shutil.py", line 254, in rmtree
    os.rmdir(path)
OSError: [Errno 39] Directory not empty: 'psadata/fitted_trajs'
-------------------- >> begin captured logging << --------------------
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk_dims.dcd...
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk_dims2.dcd...
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk_dims.dcd...
MDAnalysis.analysis.align: INFO: RMS-fitting on 214 atoms.
MDAnalysis.analysis.align: INFO: Wrote 98 RMS-fitted coordinate frames to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa001.dcd'
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa001.dcd...
MDAnalysis.analysis.align: INFO: RMS-fitting on 214 atoms.
MDAnalysis.analysis.align: INFO: Wrote 102 RMS-fitted coordinate frames to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa002.dcd'
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa002.dcd...
MDAnalysis.analysis.align: INFO: RMS-fitting on 214 atoms.
MDAnalysis.analysis.align: INFO: Wrote 98 RMS-fitted coordinate frames to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa003.dcd'
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa003.dcd...
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/hausdorff'.npz
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/hausdorff'.dat
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/discrete_frechet'.npz
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/discrete_frechet'.dat
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: test_reversal_frechet (MDAnalysisTests.analysis.test_psa.TestPSAnalysis)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/analysis/test_psa.py", line 57, in tearDown
    shutil.rmtree('psadata') # remove psa data directory
  File "/usr/remote/python/2.7.8/lib/python2.7/shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "/usr/remote/python/2.7.8/lib/python2.7/shutil.py", line 256, in rmtree
    onerror(os.rmdir, path, sys.exc_info())
  File "/usr/remote/python/2.7.8/lib/python2.7/shutil.py", line 254, in rmtree
    os.rmdir(path)
OSError: [Errno 39] Directory not empty: 'psadata/fitted_trajs'
-------------------- >> begin captured logging << --------------------
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk_dims.dcd...
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk_dims2.dcd...
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk_dims.dcd...
MDAnalysis.analysis.align: INFO: RMS-fitting on 214 atoms.
MDAnalysis.analysis.align: INFO: Wrote 98 RMS-fitted coordinate frames to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa001.dcd'
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa001.dcd...
MDAnalysis.analysis.align: INFO: RMS-fitting on 214 atoms.
MDAnalysis.analysis.align: INFO: Wrote 102 RMS-fitted coordinate frames to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa002.dcd'
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa002.dcd...
MDAnalysis.analysis.align: INFO: RMS-fitting on 214 atoms.
MDAnalysis.analysis.align: INFO: Wrote 98 RMS-fitted coordinate frames to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa003.dcd'
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa003.dcd...
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/hausdorff'.npz
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/hausdorff'.dat
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/discrete_frechet'.npz
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/discrete_frechet'.dat
--------------------- >> end captured logging << ---------------------

======================================================================
ERROR: test_reversal_hausdorff (MDAnalysisTests.analysis.test_psa.TestPSAnalysis)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/analysis/test_psa.py", line 57, in tearDown
    shutil.rmtree('psadata') # remove psa data directory
  File "/usr/remote/python/2.7.8/lib/python2.7/shutil.py", line 247, in rmtree
    rmtree(fullname, ignore_errors, onerror)
  File "/usr/remote/python/2.7.8/lib/python2.7/shutil.py", line 256, in rmtree
    onerror(os.rmdir, path, sys.exc_info())
  File "/usr/remote/python/2.7.8/lib/python2.7/shutil.py", line 254, in rmtree
    os.rmdir(path)
OSError: [Errno 39] Directory not empty: 'psadata/fitted_trajs'
-------------------- >> begin captured logging << --------------------
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk_dims.dcd...
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk_dims2.dcd...
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk_dims.dcd...
MDAnalysis.analysis.align: INFO: RMS-fitting on 214 atoms.
MDAnalysis.analysis.align: INFO: Wrote 98 RMS-fitted coordinate frames to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa001.dcd'
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa001.dcd...
MDAnalysis.analysis.align: INFO: RMS-fitting on 214 atoms.
MDAnalysis.analysis.align: INFO: Wrote 102 RMS-fitted coordinate frames to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa002.dcd'
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa002.dcd...
MDAnalysis.analysis.align: INFO: RMS-fitting on 214 atoms.
MDAnalysis.analysis.align: INFO: Wrote 98 RMS-fitted coordinate frames to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa003.dcd'
MDAnalysis.topology.PSF: DEBUG: PSF file /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/data/adk.psf: format STANDARD
MDAnalysis.core.AtomGroup: DEBUG: Universe.load_new(): loading /users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/fitted_trajs/fitted_psa003.dcd...
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/hausdorff'.npz
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/hausdorff'.dat
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/discrete_frechet'.npz
MDAnalysis.analysis.psa: INFO: Wrote distance matrix to file '/users/ardita/MDAPCA/mdanalysis/testsuite/MDAnalysisTests/psadata/distance_matrices/discrete_frechet'.dat
--------------------- >> end captured logging << ---------------------

----------------------------------------------------------------------
Ran 3360 tests in 695.945s

FAILED (SKIP=5, errors=3)
```

Could anyone please explain the difference between nosetests and ./mda_nosetests here? Are they performing the same tests. Thanks.

**Comment by ashkurti** [comment]
Maybe the number of tests and the number of failures changes after each try due to some default execution walltime of the tests? Thanks.

**Comment by richardjgowers** [comment]
Hi

You're correct about timing out, if you're using parallel testing you need to add `--process-timeout=120`

And yeah, just go ahead with the pull request you originally wanted to do

**Comment by orbeckst** [comment]
Definitely do a PR – it gets tested on Travis automatically and we typically look at the Travis report and then start code reviews with those test results in mind.

Do I understand you correctly in that the tests related to the `float32` format in the NCDF writer are now passing, i.e. do you get all passes for
```
./mda_nosetstests -v ./coordinates/test_netcdf.py
```
i.e., did c2048578024381890530b609ab5050f47e7527a4 fix the issue that you originally posted?

Thanks for keeping testing – it isn't usually that complicated but we had #556 at the same time, which made it difficult to entangle everything.

**Comment by ashkurti** [comment]
The netcdf tests do pass. But I have hundreds of other tests that do not pass now ... so despite this as you say I will go ahead with the PR ...

**Comment by orbeckst** [comment]
Thanks for the feedback on the NCDFWriter. That's good.

Do you get hundreds of errors with
```
./mda_nosetests -v --processes=8 --process-timeout=120 --with-memleak
```
too?

**Comment by ashkurti** [comment]
I just finished retesting with
```
./mda_nosetests -v --processes=4 --process-timeout=120 --with-memleak
```
and get: FAILED (KNOWNFAIL=3, SKIP=2, errors=3)

I think these are related with the PSA tests.

**Comment by orbeckst** [comment]
Ok, that counts as success ;-). Thanks for checking.

I am closing this issue and @sseyler will have to sort out #557.

I look forward to your PR. Code review will happen on the PR.

**Comment by ashkurti** [comment]
After logging off my workstation, I have restarted a similar session as before using the virtualenv I created on purpose for MDAnalysis. However, I think I am missing something since I cannot get MDAnalysis to be imported. Has any of you ever experienced such issues as those I do, that are logged here https://gist.github.com/ashkurti/0be0b4946d0b4a34d653? 

Any suggestions will be appreciated, thanks :)

**Comment by kain88-de** [comment]
This are the same errors that travis is giving us in #560.  I'm not sure why that is happening. One possible explanation could be that we have introduced a cyclic-important  recently. 

Out of curiosity could you list the packages installed in your virtual env with 'pip freeze'

**Comment by ashkurti** [comment]
@kain88-de here you go with the packages installed in my virtualenv:
```
[MDAPCAvenv] ardita@moriarty 334% pip freeze
Cython==0.23.4
GridDataFormats==0.3.0
MDAnalysis==0.13.0-dev0
MDAnalysisTests==0.13.0-dev0
biopython==1.66
decorator==4.0.4
netCDF4==1.0.7
networkx==1.10
nose==1.3.7
numpy==1.10.1
scipy==0.16.1
six==1.10.0
tempdir==0.6
wsgiref==0.1.2
```

Are you thinking of strategy to eliminate errors yet ... please let me know ...

**Comment by ashkurti** [comment]
I see that you have closed Issue #560 that means that somehow you did address the problems. Could you please let me know what did you do - I am quite spending a lot of time on figuring out these kind of errors than on dealing with the code itself ...

**Comment by kain88-de** [comment]
I didn't close the Pull Request. I just referenced it from **this** closed issue and github is showing that. I'm not sure where the solution is though since I can't reproduce it on my machine. I'll try to reproduce the error then I can go fixing it. Btw when you try to fix it. instead of running the full testsuite just load the package into python and examine the stack strace.

**Comment by ashkurti** [comment]
```
[MDAPCAvenv] ardita@moriarty 335% python 
Python 2.7.8 (default, Nov  6 2014, 13:37:40) 
[GCC 4.4.7 20120313 (Red Hat 4.4.7-11)] on linux2
Type "help", "copyright", "credits" or "license" for more information.
>>> import MDAnalysis as mda
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/users/ardita/MDAPCAvenv/lib/python2.7/site-packages/MDAnalysis-0.13.0_dev0-py2.7-linux-x86_64.egg/MDAnalysis/__init__.py", line 151, in <module>
    from .lib import log
  File "/users/ardita/MDAPCAvenv/lib/python2.7/site-packages/MDAnalysis-0.13.0_dev0-py2.7-linux-x86_64.egg/MDAnalysis/lib/__init__.py", line 30, in <module>
    from . import distances  # distances relies on mdamath
  File "/users/ardita/MDAPCAvenv/lib/python2.7/site-packages/MDAnalysis-0.13.0_dev0-py2.7-linux-x86_64.egg/MDAnalysis/lib/distances.py", line 68, in <module>
    package="MDAnalysis.lib")
  File "/usr/remote/python/2.7.8/lib/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
ImportError: dynamic module does not define init function (init_distances)
```

**Comment by ashkurti** [comment]
@kain88-de - you are right, it is still open.

**Comment by richardjgowers** [comment]
`(dynamic module does not define init function (init_distances))` - this error is caused by the Cython not importing properly.

How had you installed MDA in the virtual env? (I'm guessing like here? https://github.com/MDAnalysis/mdanalysis/wiki/Setup-Development-Environment )

If you rerun `pip install -e package/` does this fix it?

**Comment by ashkurti** [comment]
Yes - installed MDA like you say;
No - rerunning ``` pip install -e package ``` does not fix this...

I have also created a new virtualenv and installed everything from scratch but get the same errors :- ( Something might have changed from last week?

**Comment by ashkurti** [comment]
Or any ideas on any environment variables related to Cython?

**Comment by ashkurti** [comment]
Guess what? I checked back to the commit at ```https://github.com/MDAnalysis/mdanalysis/commit/0c3aa90cf0a748642a9aab5e50562ca1b9fedc18  ``` and tests are working again except for 5 as I have documented at https://gist.github.com/ashkurti/1703148a679da045f48f ... 

Would you guys @orbeckst et al mind if I do my pull request from that commit then, although it is 2 weeks old, sorry about that...

**Comment by richardjgowers** [comment]
Yeah that'd be fine

**Comment by hainm** [comment]
Few things

+ try to use conda for managing env
+ try to delete all installs for mda and make a fresh install

**Comment by hainm** [comment]
I used to get similar error but it is because I accidentally add python2 path to my python3 env

**Comment by ashkurti** [comment]
@hainm - Thanks for your tips and please consider the following:
* I do not use conda as a point of principle so as I have more control myself on what has been installed and how has been installed. Plus a large number of users of the packages I have been developing do not use conda and I get questions related with separate packages installation issues, so I will need to face them myself first.
* I am using virtualenv.
* I have already made several fresh installs - In fact, I have spent an enormous amount of hours last week and this week trying out non-working tests. Things do not work and fail with the same errors.
* I have not got python3 at all.
* I am now working for my PR that should be ready from the commit that I indicated previously, hope this does not create problems for anyone :-)

**Comment by hainm** [comment]
@ashkurti 

at this moment, I think you should focus on your PR, creating  your own test. And if your change passed travis build, it should be fine.

any way, have you tried to `rm -r build` directory yet?

**Comment by kain88-de** [comment]
So I think I found the import error. Looks like cython expects that source files and modules have the same name and location. I'll do some more tests on it.

**Comment by kain88-de** [comment]
@ashkurti try it again with the recent master version. the last fails should be fixed now.

**Comment by orbeckst** [comment]
On 7 Dec, 2015, at 15:42, kain88-de wrote:

> @ashkurti try it again with the recent master version. the last fails should be fixed now.

Where "master" is the develop branch...

--
Oliver Beckstein * orbeckst@gmx.net
skype: orbeckst  * orbeckst@gmail.com

**Comment by ashkurti** [comment]
Hi, what would be the suggested way to do pull requests. Do I have writing permissions into MDAnalysis so as to push a specific branch into it. If not, what are the alternatives.

Thanks!!

**Comment by richardjgowers** [comment]
Hi

You should be able to fork this repo (button in the top right), which makes you your own copy.  You can then push your new branch to your fork, then create a PR from that branch in your repo to our develop branch here.

**Comment by ashkurti** [comment]
Ok, I have done every bit, a part from the ```create a PR from that branch in your repo to our develop branch here``` bit. Should I use for this the hub wrapper, or should I upload my forked repository to my github account and then send the PR vi web from there?

**Comment by hainm** [comment]
Hi

Here is the tutorial https://help.github.com/articles/using-pull-requests/

**Comment by orbeckst** [comment]
On 8 Dec, 2015, at 10:23, ashkurti wrote:

> Ok, I have done every bit, a part from the create a PR from that branch in your repo to our develop branch here bit. Should I use for this the hub wrapper, or should I upload my forked repository to my github account and then send the PR vi web from there?

Through the web:

1. push your changes to your fork on GitHub
2. create PR

This PR will then show up in MDAnalysis under https://github.com/MDAnalysis/mdanalysis/pulls and folks will then look at it. The tests will be run automatically. Folks will then look at it and comment. Monitor the PR and have a discussion there.

Push additions to YOUR repo on GitHub . They will show up under the PR.

Then more commenting and discussion until everyone is happy, tests pass, and someone will merge into develop.

--
Oliver Beckstein * orbeckst@gmx.net
skype: orbeckst  * orbeckst@gmail.com

**Comment by ashkurti** [comment]
Thanks @orbeckst and @hainm for the details, I have just sent the PR. Looking forward for the feedback :-) and will continue any discussions at the automatic issue open for the PR.
